src/transformers/trainer.py

```diff
@@ -805,20 +800,6 @@ def _get_train_sampler(self) -> Optional[torch.utils.data.Sampler]:
         if self.train_dataset is None or not has_length(self.train_dataset):
             return None
 
-        generator = None
-        if self.args.world_size <= 1:
-            generator = torch.Generator()
-            # for backwards compatibility, we generate a seed here (which is sampled from a generator seeded with
-            # `args.seed`) if data_seed isn't provided.
-            # Further on in this method, we default to `args.seed` instead.
-            if self.args.data_seed is None:
-                seed = int(torch.empty((), dtype=torch.int64).random_().item())
-            else:
-                seed = self.args.data_seed
-            generator.manual_seed(seed)
-
-        seed = self.args.data_seed if self.args.data_seed is not None else self.args.seed
```

Comment:
Nope, Accelerate handles that better and doing this actually hinders reproducibility in that sense. (We use an entirely different dataloader/sampler mechanic than the one used here, so our version of seed isn't the same as what's being used for the samplers/dataloaders here)

Reply:
The *tests* haven't shown it to, because (from what Sylvain and I were able to find), our dataloaders don't play with the same random seed setting that these do/applying generators. Keeping this in as a generator also causes tests to fail, showing that it's detrimental in that sense now with our new DataLoader setup. 

Also it's used when `__iter__` through the dataset here: https://github.com/huggingface/transformers/blob/main/src/transformers/trainer_pt_utils.py#L665-L667. Because we don't use a torch `DistributedSampler`, I don't believe it applies to us here. 

If later down the road someone points to a reproducible issue to this, I'll review it again but in my deep dive last week I didn't find an issue.